cli/module_graph2.rs

```diff
@@ -1393,13 +1397,8 @@ impl GraphBuilder2 {
   /// Move out the graph from the builder to be utilized further.  An optional
   /// lockfile can be provided, where if the sources in the graph do not match
   /// the expected lockfile, an error will be logged and the process will exit.
-  ///
-  /// TODO(@kitsonk) this should really be owned by the graph, but currently
-  /// the lockfile is behind a mutex in program_state, which makes it really
-  /// hard to not pass around as a reference, which if the Graph owned it, it
-  /// would need lifetime parameters and lifetime parameters are 😭
-  pub fn get_graph(self, maybe_lockfile: &Option<Mutex<Lockfile>>) -> Graph2 {
-    self.graph.lock(maybe_lockfile);
+  pub fn get_graph(self) -> Graph2 {
+    self.graph.lock(self.maybe_lockfile.as_ref());
```

Comment:
Why not pass it to the graph itself in `::new()` like the `handler`?  Then `self.graph.lock` and just `self.maybe_lockfile` itself.

Reply:
Done